thunder/torch/__init__.py

```diff
@@ -3590,253 +3576,191 @@ def cross_entropy(
         lambda: f"Deprecated size_average={size_average} and reduce={reduce} is not supported!",
     )
 
+    _cross_entropy_input_checks(a, target, weight, ignore_index, reduction, label_smoothing)
+
+    # channels dimension is either the first one if no batch dim present (i.e. a.shape[0]),
+    # or right next to it (i.e. a.shape[1]).
+    channels_dim = 1 if a.ndim >= 2 else 0
+
+    # NOTE This short-circuit is subject to change and is placed ahead of other input checks to match PyTorch behavior.
+    # The expected behavior when the target and input have zero elements:
+    #   reduction = 'none' --- tensor([], shape) or tensor(0.)
+    #   reduction = 'sum'  --- tensor(0.)
+    #   reduction = 'mean' --- tensor(nan)
+    # Mean reduction on empty tensors produces NaN.
+    if a.numel() == 0:
+        if reduction == "none":
```

Comment:
Note to myself. Another justification for op-based constraint. `reduction` string needs to be baked-in and properly checked in prologue trace.